genie/process_mutation.py

```diff
@@ -226,18 +230,163 @@ def process_mutation_workflow(
+        namedtuple: tuple with all the paths
+    """
+    input_files_dir = tempfile.mkdtemp(dir=workdir)
+    output_files_dir = tempfile.mkdtemp(dir=workdir)
+    Filepaths = namedtuple(
+        "Filepaths",
+        [
+            "input_files_dir",
+            "output_files_dir",
+            "error_dir",
+            "merged_maf_path",
+            "full_maf_path",
+            "narrow_maf_path",
+            "full_error_report_path",
+        ],
+    )
+    annotation_paths = Filepaths(
+        input_files_dir=input_files_dir,
+        output_files_dir=output_files_dir,
+        error_dir=os.path.join(output_files_dir, f"{center}_error_reports"),
+        merged_maf_path=os.path.join(
+            output_files_dir, f"data_mutations_extended_{center}.txt"
+        ),
+        full_maf_path=os.path.join(
+            workdir, center, "staging", f"data_mutations_extended_{center}.txt"
+        ),
+        narrow_maf_path=os.path.join(
+            workdir,
+            center,
+            "staging",
+            f"data_mutations_extended_{center}_MAF_narrow.txt",
+        ),
+        full_error_report_path=os.path.join(
+            workdir,
+            center,
+            "staging",
+            f"failed_annotations_error_report.txt",
+        ),
+    )
+    return annotation_paths
+
+
+def concat_annotation_error_reports(
+    center: str,
+    input_dir: str,
+) -> pd.DataFrame:
+    """Concatenates the annotation error reports
+
+    Args:
+        center (str): name of center associated with error report
+        input_dir (str): directory where error reports are
+    Returns:
+        pd.DataFrame: full annotation error report
+    """
+    error_files = os.listdir(input_dir)
+    chunk_size = 10000
+    error_reports = []
+
+    # Read and concatenate TSV files in chunks
+    for file in error_files:
+        for chunk in pd.read_csv(
+            os.path.join(input_dir, file), sep="\t", chunksize=chunk_size
+        ):
+            error_reports.append(chunk)
+    full_error_report = pd.concat(error_reports)
+    full_error_report["Center"] = center
+    return full_error_report
+
+
+def check_annotation_error_reports(
+    syn: Synapse, maf_table_synid: str, full_error_report: pd.DataFrame, center: str
+) -> None:
+    """A simple QC check to make sure our genome nexus error report
+       failed annotations matches our final processed maf table's failed
+       annotations
+
+    Args:
+        syn (Synapse): synapse client
+        maf_table_synid (str): synapse_id of the narrow maf table
+        full_error_report (pd.DataFrame): the failed annotations error report
+        center (str): the center this is for
+    """
+    maf_table_df = extract.get_syntabledf(
+        syn=syn,
+        query_string=(
+            f"SELECT * FROM {maf_table_synid} "
+            f"WHERE Center = '{center}' AND "
+            "Annotation_Status = 'FAILED'"
+        ),
+    )
+    assert len(maf_table_df) == len(full_error_report), (
```

Comment:
@rxu17 this is a great idea, but also a tiny bit risky for processing itself because mutation processing takes days.  I wonder if we want to log this instead of throwing an error that would pause processing all together

Reply:
Yes created here: [GEN-1081](https://sagebionetworks.jira.com/browse/GEN-1081)